edu-services/gradebook-service/impl/src/java/org/sakaiproject/component/gradebook/GradebookServiceHibernateImpl.java

```diff
@@ -2112,6 +2112,11 @@ public Object doInHibernate(Session session) throws HibernateException {
 
 			  String studentId = gradeDef.getStudentUid();
 
+              // use the grader ID from the definition if it is not null, otherwise use the current user ID
+              String graderUid = gradeDef.getGraderUid() != null ? gradeDef.getGraderUid() : graderId;
+              // use the grade date from the definition if it is not null, otherwise use the current date
+              Date gradedDate = gradeDef.getDateRecorded() != null ? gradeDef.getDateRecorded() : now;
+
```

Comment:
Ugh, yeah... I changed them to gross, nasty, ugly tabs + spaces. I left that weird tab + space indentation to match the rest.